pkg/provider/electricity_map.go

```diff
@@ -61,14 +61,11 @@ func (e *ElectricityMapClient) GetCarbonIntensity(ctx context.Context, location
 		return nil, err
 	}
 
-	validFrom, err := time.Parse(time.RFC3339Nano, data.UpdatedAt)
-	if err != nil {
-		return nil, err
-	}
-	validTo, err := time.Parse(time.RFC3339Nano, data.DateTime)
+	validFrom, err := time.Parse(time.RFC3339Nano, data.DateTime)
 	if err != nil {
 		return nil, err
 	}
+	validTo := validFrom.Add(60 * time.Minute)
```

Comment:
This was wrong datetime is the start of the period and the data is hourly.

Reply:
Is the data always hourly?

I know that with some places data is shared at a higher resolution than hourly (Taiwan and Australia publish data in 15 minute increments, I think), and the UK has 48 30 minute settlement periods in a given day.

Reply:
Yes we need to check this!  The docs are not very clear. https://static.electricitymaps.com/api/docs/index.html#live-carbon-intensity

However getting the forecast for any zone including UK the data is always hourly.

```
 curl -s -H 'auth-token:xxxxx' 'https://api-access.electricitymaps.com/2w97h07rvxvuaa1g/carbon-intensity/lates    t?zone=GB' | jq .

{
  "zone": "GB",
  "forecast": [
    {
      "carbonIntensity": 191,
      "datetime": "2023-07-13T11:00:00.000Z"
    },
    {
      "carbonIntensity": 191,
      "datetime": "2023-07-13T12:00:00.000Z"
    },
    {
      "carbonIntensity": 211,
      "datetime": "2023-07-13T13:00:00.000Z"
    },
   ...
  "updatedAt": "2023-07-13T10:50:56.800Z"
}
```

My suggestion is we go with this but I'll open a support request and confirm with them the data is hourly?

Reply:
Good idea, @rossf7. 

I'll assume no action is needed at my end for this for the time being. Thanks for the extra go links too  👍

Reply:
Hi, just saw this PR (🙌) and I can confirm that everything is hourly - I'll also make sure that we make this more clear in the documentation :)